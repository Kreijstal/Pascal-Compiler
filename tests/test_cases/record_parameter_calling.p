program TestRecordParameters;
type
  TPoint = record x: integer; end;
var
  myPoint: TPoint;
procedure ModifyCopy(pt: TPoint);
begin
  pt.x := 999;
  writeln('Inside ModifyCopy, pt.x = ', pt.x);
end;
procedure ModifyOriginal(var pt: TPoint);
begin
  pt.x := pt.x + 1;
end;
begin
  myPoint.x := 10;
  writeln('Before calls, myPoint.x = ', myPoint.x);
  ModifyCopy(myPoint);
  writeln('After ModifyCopy, myPoint.x = ', myPoint.x);
  ModifyOriginal(myPoint);
  writeln('After ModifyOriginal, myPoint.x = ', myPoint.x);
end.
