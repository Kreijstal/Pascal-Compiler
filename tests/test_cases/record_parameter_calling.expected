Before calls, myPoint.x = 10
Inside ModifyCopy, pt.x = 999
After ModifyCopy, myPoint.x = 10
After ModifyOriginal, myPoint.x = 11
